From 171aaa84047acc3b1ed99cd7c1d8c7a5b0394b10 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 31 Jan 2018 13:06:48 -0500
Subject: [PATCH 46/60] grubby: Make sure configure$BOOTLOADER variables
are
 set correctly.

When we've chosen a bootloader because it's default for a platform, and
we've already determined it's not overridden by the command line, set
the configure$BOOTLOADER variable to 1 so that our checks for which
bootloader are selected work correctly.

Resolves: rhbz#1340893

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grubby.c | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/grubby.c b/grubby.c
index c2fab23..1d3e924 100644
--- a/grubby.c
+++ b/grubby.c
@@ -4659,22 +4659,27 @@ int main(int argc, const char ** argv) {
     if (!cfi) {
         if (grub2FindConfig(&grub2ConfigType)) {
 	    cfi = &grub2ConfigType;
+	    configureGrub2 = 1;
 	    if (envPath)
 		cfi->envFile = envPath;
-        } else
+        } else {
       #ifdef __ia64__
 	    cfi = &eliloConfigType;
-      #elif __powerpc__
+	    configureLilo = 1;
+      #elif defined(__powerpc__)
 	    cfi = &yabootConfigType;
-      #elif __sparc__
+	    configureYaboot = 1;
+      #elif defined(__sparc__)
             cfi = &siloConfigType;
-      #elif __s390__
+	    configureSilo = 1;
+      #elif defined(__s390__) || defined(__s390x__)
             cfi = &ziplConfigType;
-      #elif __s390x__
-            cfi = &ziplConfigtype;
+	    configureZipl = 1
       #else
 	    cfi = &grubConfigType;
+	    configureGrub = 1;
       #endif
+	}
     }
 
     if (!grubConfig) {
-- 
2.19.1

